tracing/trace.go

```diff
@@ -0,0 +1,174 @@
+// Package tracing supplies access to tracer providers.
+//
+// It uses the [otel](https://opentelemetry.io/docs/instrumentation/go/getting-started/) tracing library, connecting
+// to an OpenTelemetry collector.
+//
+// The package relies on, and is meant to be used in conjunction with the observability package
+package tracing
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/exporters/stdout/stdouttrace"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/trace"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+const (
+	instrumentationCodeProviderName = "github.com/els0r/goProbe/pkg/tracing"
+)
+
+type tracingConfig struct {
+	exporter sdktrace.SpanExporter
+}
+
+// Option allows to configure the tracing setup
+type Option func(*tracingConfig) error
+
+// WithGRPCExporter sets up an exporter using a gRPC connection to the trace collector
+func WithGRPCExporter(ctx context.Context, collectorEndpoint string) Option {
+	return func(tc *tracingConfig) error {
+		if collectorEndpoint == "" {
+			return fmt.Errorf("no collector endpoint set")
```

Comment:
Not a formatted error, recommend using `errors.New()` and define a reusable variable...